selftest/test_images.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env python3
+
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for license information.
+# --------------------------------------------------------------------------
+
+"""Azure VM utilities self-tests script."""
+
+import os
+import shlex
+import subprocess
+import time
+import uuid
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import List
+
+import pytest
+
+from . import selftest
+
+# pylint: disable=unknown-option-value
+# pylint: disable=line-too-long
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-positional-arguments
+# pylint: disable=unused-argument
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=redefined-outer-name
+
+DEFAULT_GEN1_VM_SIZES = [
+    sku.vm_size
+    for sku in selftest.SKU_CONFIGS.values()
+    if sku.vm_size_type == "x64" and not sku.vm_size.endswith("v6")
+]
+
+DEFAULT_GEN2_VM_SIZES = [
+    sku.vm_size for sku in selftest.SKU_CONFIGS.values() if sku.vm_size_type == "x64"
+]
+
+DEFAULT_ARM64_VM_SIZES = [
+    sku.vm_size for sku in selftest.SKU_CONFIGS.values() if sku.vm_size_type == "arm64"
+]
+
+CUSTOM_IMAGES = [
+    image for image in os.getenv("TEST_CUSTOM_IMAGES", "").split(",") if image
+]
+ENV_TEST_CUSTOM_VM_SIZES = os.getenv("TEST_CUSTOM_VM_SIZES", "")
+if ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_GEN1_VM_SIZES":
+    CUSTOM_VM_SIZES = DEFAULT_GEN1_VM_SIZES
+elif ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_GEN2_VM_SIZES":
+    CUSTOM_VM_SIZES = DEFAULT_GEN2_VM_SIZES
+elif ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_ARM64_VM_SIZES":
+    CUSTOM_VM_SIZES = DEFAULT_ARM64_VM_SIZES
+else:
+    CUSTOM_VM_SIZES = [image for image in ENV_TEST_CUSTOM_VM_SIZES.split(",") if image]
+
+
+def subprocess_run(cmd: List[str], check: bool = True):
+    """Run a subprocess command and capture outputs as utf-8."""
+    print(f"executing command: {shlex.join(cmd)}")
+    try:
+        proc = subprocess.run(
+            cmd,
+            check=check,
+            capture_output=True,
+            text=True,
+            encoding="utf-8",
+        )
+    except subprocess.CalledProcessError as error:
+        print(
+            f"error running command: {error} stdout={error.stdout} stderr={error.stderr}"
+        )
+        raise
+
+    print(f"executed command: {proc}")
+    return proc
+
+
+@pytest.fixture
+def ssh_key_path(tmp_path):
+    """Generate a temporary SSH key pair for the test."""
+    path = tmp_path / "id_rsa"
+    subprocess.run(
+        ["ssh-keygen", "-t", "rsa", "-b", "2048", "-f", str(path), "-N", ""], check=True
+    )
+    yield path
```

Comment:
Why the yields for all the fixtures?

Reply:
not that it matters for these, but generally fixtures "yield" and then have the opportunity to cleanup/teardown after.  though a return would be same in this case as no cleanup is required